examples/temperature-measurement-app/esp32/main/diagnostic-logs-provider-delegate-impl.cpp

```diff
@@ -95,34 +91,17 @@ size_t LogProvider::GetCrashSize()
     size_t outSize = 0;
 
 #if defined(CONFIG_ESP_COREDUMP_ENABLE_TO_FLASH) && defined(CONFIG_ESP_COREDUMP_DATA_FORMAT_ELF)
-    size_t unusedOutAddr;
-    esp_err_t esp_err = esp_core_dump_image_get(&unusedOutAddr, &outSize);
+    // Verify that the crash is present and sane
+    size_t unusedOutAddr, unusedOutSize;
+    esp_err_t esp_err = esp_core_dump_image_get(&unusedOutAddr, &unusedOutSize);
     VerifyOrReturnValue(esp_err == ESP_OK, 0, ChipLogError(DeviceLayer, "Failed to get core dump image, esp_err:%d", esp_err));
+
+    outSize = sizeof(esp_core_dump_summary_t);
```

Comment:
Need to check that the outsize from `esp_core_dump_image_get` is `>= sizeof(esp_core_dump_summary_t)` rather than ignoring the outsize.

Reply:
`esp_core_dump_image_get()` gets the complete core dump image and  outSize will always be greater than `sizeof(esp_core_dump_summary_t)`, and the check you suggested to add does make sense. But, as the core dump image size and summary size is not related, I used a different API, [`esp_core_dump_image_check()`](https://github.com/espressif/esp-idf/blob/master/components/espcoredump/include/esp_core_dump.h#L95-L106) here which also checks the integrity along with size checks as well.

Reply:
resolving the thread!